metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishedRecordDao.java

```diff
@@ -147,59 +150,107 @@ long countDepublishedRecordsForDataset(String datasetId) {
    * @param searchQuery Search query for the record ID. Can be null.
    * @return A (possibly empty) list of depublished records.
    */
   public List<DepublishedRecordView> getDepublishedRecords(String datasetId, int page,
-          DepublishedRecordSortField sortField, SortDirection sortDirection, String searchQuery) {
-
-    // Create query.
-    final Query<DepublishedRecord> query = morphiaDatastoreProvider.getDatastore()
-            .createQuery(DepublishedRecord.class);
-    query.field(DepublishedRecord.DATASET_ID_FIELD).equal(datasetId);
-    if (StringUtils.isNotBlank(searchQuery)) {
-      query.field(DepublishedRecord.RECORD_ID_FIELD).contains(searchQuery);
-    }
-
-    // Set ordering
-    query.order(sortDirection.createSort(sortField.getDatabaseField()));
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
+    final Query<DepublishedRecord> query = prepareQueryForDepublishedRecords(datasetId, sortField,
+        sortDirection, depublicationState, searchQuery);
 
     // Compute pagination
     final int skip = page * pageSize;
     final int limit = skip + pageSize;
     final FindOptions findOptions = new FindOptions().skip(skip).limit(limit);
 
     // Execute query with correct pagination
-    final List<DepublishedRecord> result = ExternalRequestUtil
-            .retryableExternalRequestConnectionReset(() -> {
-              try (final MorphiaCursor<DepublishedRecord> cursor = query.find(findOptions)) {
-                return cursor.toList();
-              }
-            });
+    final List<DepublishedRecord> result = getListOfQuery(query, findOptions);
 
     // Convert result to right object.
     return result.stream().map(DepublishedRecordView::new).collect(Collectors.toList());
   }
 
   /**
-   * This method marks all depublished records of a given dataset as not depublished (and removes
-   * the depublished date).
+   * Get all depublished records for a given dataset.
+   * <p>This method is to be used with caution sicne it doesn't have a limit on the returned items.
+   * It is mainly used for minimize internal to the application database requests.</p>
    *
-   * @param datasetId The dataset for which to do this.
+   * @param datasetId The dataset for which to retrieve the records. Cannot be null.
+   * @param sortField The sorting field. Cannot be null.
+   * @param sortDirection The sorting direction. Cannot be null.
+   * @param depublicationState The depublication state of the records. Can be null.
+   * @param searchQuery Search query for the record ID. Can be null.
+   * @return A (possibly empty) list of depublished records.
    */
-  public void markAllAsNotDepublished(String datasetId){
+  public Set<String> getAllDepublishedRecords(String datasetId,
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
```

Comment:
And this method is never called with a non-null searchQuery.

Reply:
I thought I'd be generic in both cases, but fair enough I'll remove the parameter